src/GitTfs.VsCommon/TfsHelper.Vs2017Base.cs

```diff
@@ -61,7 +72,51 @@ protected override bool HasWorkItems(Changeset changeset)
         /// </returns>
         protected string GetVsInstallDir()
         {
-            return @"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise";
+            if (myVisualStudioInstallationPath != null)
+                return myVisualStudioInstallationPath;
+
+            var setupConfiguration = (ISetupConfiguration2)GetSetupConfiguration();
+            IEnumSetupInstances instancesEnumerator = setupConfiguration.EnumAllInstances();
+
+            int fetched;
+            var instances = new ISetupInstance[1];
+            while (true)
+            {
+                instancesEnumerator.Next(1, instances, out fetched);
+                if (fetched <= 0)
+                    break;
+
+                var instance = (ISetupInstance2)instances[0];
+                if (!Version.TryParse(instance.GetInstallationVersion(), out Version version))
+                {
+                    Trace.TraceError("Failed to retrieve version. Skipping VS instance.");
+                    continue;
+                }
+
+                if (version.Major != myMajorVersion)
+                {
+                    continue;
+                }
+
+                if (myVisualStudioInstallationPath != null)
+                {
+                    Trace.TraceWarning("Already found a Visual Studio version. Ignoring version at {0}", instance.GetInstallationPath());
+                    continue;
+                }
+
+                var state = instance.GetState();
+                if (state.HasFlag(InstanceState.Local) && state.HasFlag(InstanceState.Registered))
+                {
+                    myVisualStudioInstallationPath = instance.GetInstallationPath();
+                    Trace.TraceInformation("Found matching Visual Studio version at {0}", myVisualStudioInstallationPath);
+                }
+                else
+                {
+                    Trace.TraceWarning("Ingoring incomplete Visual Studio version at {0}", instance.GetInstallationPath());
```

Comment:
Nit: typo on "Ingoring" -> "Ignoring".

Reply:
Typo fixed in the seprate commit 85cba621b35e60dacf605efc2ed511a8f5487751